nix/src/main/java/org/gnode/nix/File.java

```diff
@@ -0,0 +1,201 @@
+package org.gnode.nix;
+
+import org.bytedeco.javacpp.IntPointer;
+import org.bytedeco.javacpp.Loader;
+import org.bytedeco.javacpp.Pointer;
+import org.bytedeco.javacpp.annotation.*;
+
+import java.util.ArrayList;
+
+@Platform(value = "linux",
+        include = {"<nix/File.hpp>"},
+        link = {"nix"},
+        linkpath = {"/usr/local/lib/"},
+        includepath = {"/usr/local/include/"})
```

Comment:
Please find a solution for setting `value`, `linkpath` and `includepath` more in the build conf or via environment variables. Or file an issue for M1 and label it as `high priority`.